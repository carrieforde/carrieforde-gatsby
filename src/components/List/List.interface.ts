import { ListType } from 'entities/list-type.type';

export interface ListProps {
  type: ListType;
  className?: string;
}

export interface ListItemProps {
  className?: string;
}
