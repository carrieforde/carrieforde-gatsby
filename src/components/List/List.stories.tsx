import React from 'react';
import List, { ListItem } from './List';
import { ListProps } from './List.interface';
import { Story, Meta } from '@storybook/react';

export default {
  title: 'List',
  component: List,
} as Meta;

const Template: Story<ListProps> = (args) => <List {...args} />;

export const BulletedList = Template.bind({});
export const NumberedList = Template.bind({});

BulletedList.args = {
  type: 'BULLETED',
  children: [
    <ListItem key="Typescript">Typescript</ListItem>,
    <ListItem key="React">React</ListItem>,
    <ListItem key="HTML">HTML</ListItem>,
    <ListItem key="CSS (& Sass)">CSS (& Sass)</ListItem>,
    <ListItem key="Git">Git</ListItem>,
    <ListItem key="Firebase">Firebase</ListItem>,
  ],
};

NumberedList.args = {
  ...BulletedList.args,
  type: 'NUMBERED',
};
