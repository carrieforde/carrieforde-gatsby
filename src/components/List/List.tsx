import React, { useMemo } from "react";
import { ListItemProps, ListProps } from "./List.interface";

const List: React.FC<ListProps> = ({ type, className, children }) => {
  const ListType = useMemo(() => (type === "BULLETED" ? "ul" : "ol"), []);

  return <ListType className={className}>{children}</ListType>;
};

const ListItem: React.FC<ListItemProps> = ({ className, children }) => (
  <li className={className}>{children}</li>
);

export { List as default, ListItem };
