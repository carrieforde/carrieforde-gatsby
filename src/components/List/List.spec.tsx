import { render } from '@testing-library/react';
import React from 'react';
import { BulletedList, NumberedList } from './List.stories';

function getListComponents(container: HTMLElement) {
  return {
    bulleted: container.querySelector('ul'),
    numbered: container.querySelector('ol'),
  };
}

describe('List', () => {
  it('should render a bulleted list', () => {
    const { container } = render(<BulletedList {...BulletedList.args} />);
    const { bulleted, numbered } = getListComponents(container);

    expect(container).toMatchSnapshot();
    expect(bulleted).toBeInTheDocument();
    expect(numbered).not.toBeInTheDocument();
  });

  it('should render a numbered list', () => {
    const { container } = render(<NumberedList {...NumberedList.args} />);
    const { bulleted, numbered } = getListComponents(container);

    expect(container).toMatchSnapshot();
    expect(bulleted).not.toBeInTheDocument();
    expect(numbered).toBeInTheDocument();
  });
});
