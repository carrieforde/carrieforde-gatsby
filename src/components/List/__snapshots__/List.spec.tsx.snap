// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render a bulleted list 1`] = `
<div>
  <ul>
    <li>
      Typescript
    </li>
    <li>
      React
    </li>
    <li>
      HTML
    </li>
    <li>
      CSS (& Sass)
    </li>
    <li>
      Git
    </li>
    <li>
      Firebase
    </li>
  </ul>
</div>
`;

exports[`List should render a numbered list 1`] = `
<div>
  <ol>
    <li>
      Typescript
    </li>
    <li>
      React
    </li>
    <li>
      HTML
    </li>
    <li>
      CSS (& Sass)
    </li>
    <li>
      Git
    </li>
    <li>
      Firebase
    </li>
  </ol>
</div>
`;
